File: clients/java/src/main/java/net/weweave/tubewarder/client/KeyValue.java
package net.weweave.tubewarder.client;

import java.io.Serializable;

public class KeyValue implements Serializable {
    private String key;
    private Object value;

    public KeyValue() {

    }

    public KeyValue(String key, Object value) {
        setKey(key);
        setValue(value);
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public Object getValue() {
        return value;
    }

    public void setValue(Object value) {
        this.value = value;
    }
}

File: clients/php/KeyValue.php
<?php
namespace Tubewarder;

class KeyValue {
    private $key;
    private $value;

    public function __construct($key, $value = '') {
        $this->setKey($key);
        $this->setValue($value);
    }

    /**
     * @return string
     */
    public function getKey() {
        return $this->key;
    }

    /**
     * @param string $key
     */
    public function setKey($key) {
        $this->key = $key;
    }

    /**
     * @return mixed
     */
    public function getValue() {
        return $this->value;
    }

    /**
     * @param mixed $value
     */
    public function setValue($value) {
        $this->value = $value;
    }

    /**
     * @return array
     */
    public function getObject() {
        $o = array(
            'key' => $this->getKey(),
            'value' => $this->getValue()
        );
        return $o;
    }
}